backend_new/src/services/listing.service.ts

```diff
@@ -122,21 +131,59 @@ views.details = {
   ...views.full,
 };
 
+const CRON_JOB_NAME = 'LISTING_CRON_JOB';
+
 /*
   this is the service for listings
   it handles all the backend's business logic for reading in listing(s)
 */
 @Injectable()
-export class ListingService {
+export class ListingService implements OnModuleInit {
   constructor(
     private prisma: PrismaService,
     private translationService: TranslationService,
     private httpService: HttpService,
     private afsService: ApplicationFlaggedSetService,
     private emailService: EmailService,
     private configService: ConfigService,
+    @Inject(Logger)
+    private logger = new Logger(ListingService.name),
+    private schedulerRegistry: SchedulerRegistry,
   ) {}
 
+  onModuleInit() {
```

Comment:
This function looks basically identical to the afs cron job one. Could you pull it out to a helper function and use it in both places?

Reply:
good call, I pulled both out for a shared function, lemme know what you think